### DISCUSSION TITLE
Insufficient funds error when withdraw ETH/ERC20 from op-stack L2

### DISCUSSION THREAD

**Discussion by luleigreat** [ethereum-optimism/developers#472] [General]
When I withdraw ETH/ERC20 with optimism-sdk from an op-stack L2, an error occured at this step :
```
await messenger.waitForMessageStatus(tx.hash, optimism.MessageStatus.RELAYED)
```
Error log:
```
Error: missing revert data in call exception; Transaction reverte(data="0x",transaction={"from":"0x8Cb964b9ae5802867F58Af82F3aD79A5Dc6B672" "to":"Ox90AFE01F69Ee06aeeAca475aC695D8554a73E416","data":"0xb1b1b20960ab7bd5c43bc9bf94368f30c68d022e5a6ed7e0317745f6e1ef1ec8af5a5a18","accessList":null}, error=f"reason":"processing response error","code":"SERVER_ERROR","body": "{\"jsonrpc\":\"2.0\",\"id\":44,\"error\":{\"code\":-32000,\"message\":\"insufficient funds for gas * price + value: address 0x8Cb964b9ae5802867F58420000\"}}","error":{"code":-32000},"requestBody" : "f\"method\":\"eth_call\",\"params\".f82F3aD795A5Dc6B672 have 7271012838520091005 want 171801046086000[{\"from\":\"0x8cb964b9ae5802867f58af82f3ad795a5dc6b672\",\"to\""0x90afe01f69ee06aeeaca475ac695d8554a73e416\",\"data\":\"0xb1b1b20960ab7bd5c43bc9bf94368f30c68d022e5a6ed7e0317745f6e1ef1ec8af5a5a18\"},\"latest\"],\"iN\" :44,\"jsonrpc\":\"2.0\"?","requestMethod":"PoST","url":"https://rpc.ankr.com/eth_sepolia"},code=CALL_EXCEPTION,version=providers/5.7.2)
```

It means my account balance is insufficient when it do `eth_call` to get L1 proposed block.
I think eth_call should not check my account balance,it's just a query, and it indeed doesn't check it if I do eth_call on terminate without the from parameter.

So, how to resolve this error? The gas_price on sepolia  is high recently, so this error emerged.

**Comment by opfocus** [comment]
I think this step is not necessary: `await messenger.waitForMessageStatus(tx.hash, optimism.MessageStatus.RELAYED)`. Have you checked the finalize transaction on Sepolia Etherscan?

**Reply by luleigreat** [reply]
Thanks, In fact the optimism-sdk is now superseded by [op-viem](https://viem.sh/op-stack/),I will try it
